packages/superset-ui-core/src/chart/models/ChartPlugin.ts

```diff
@@ -95,26 +102,24 @@ export default class ChartPlugin<T extends QueryFormData = QueryFormData> extend
   }
 
   register() {
-    const { key = isRequired('config.key') } = this.config;
-    getChartMetadataRegistry().registerValue(key as string, this.metadata);
-    getChartComponentRegistry().registerLoader(key as string, this.loadChart);
-    getChartControlPanelRegistry().registerValue(key as string, this.controlPanel);
-    getChartTransformPropsRegistry().registerLoader(key as string, this.loadTransformProps);
+    const key: string = this.config.key || isRequired('config.key');
```

Comment:
Given we are all in with typescript now. Could also remove the legacy `isRequired` trick that was before the age of typescript. (No need to do in this pr).

Reply:
Noted. Let's do this when all major plugins (at their index.js file) are migrated to typescript.